src/openloco/interop/hook.cpp

```diff
@@ -212,6 +213,75 @@ namespace openloco::interop
         write_memory(address, data, sizeof(data));
     }
 
+    static void* _smallHooks;
+    static uint8_t* _offset;
+
+    static void* makeJump(uint32_t address, void* fn)
```

Comment:
`make_jump`

Reply:
This pushes the address onto the stack, before calling the function. It's used to hook all the library functions. The offset is then pointed to the new operations.

Reply:
Should we always use this instead of `write_jmp`?

Reply:
I don't think so. It would need to generate different assembly based on whether `cdecl` or `stdcall` is used. `write_jmp` currently is used for both types. This is only used for 'If lib function is used that we haven't tackled, print the address to the command line'.